env/env.cc

```diff
@@ -43,6 +43,18 @@ Status Env::LoadEnv(const std::string& value, Env** result) {
   return s;
 }
 
+Status Env::LoadEnv(const std::string& value, std::shared_ptr<Env>* guard) {
+  Status s;
+#ifndef ROCKSDB_LITE
+  assert(guard != nullptr);
+  s = ObjectRegistry::NewInstance()->NewSharedObject<Env>(value, guard);
+#else
+  (void) guard;
+  s = Status::NotSupported("Cannot load environment in LITE mode: ", value);
```

Comment:
Would that make sst_dump unusable in LITE build?
Should we fallback to default Env in LITE build?

Reply:
In LITE move, the user should only use `LoadEnv(const std::string& value, Env** result)`.  I should put `#ifndef...#else` outside this entire overloaded version.

Reply:
I'm not sure I'm following. Env::LoadEnv() is called in LDBCommand::OpenDB(), so LITE build will always call it and by default it will fail. Did I miss anything?

Reply:
Or, maybe combined version
```
Status Env::LoadEnv(const std::string&, Env** result, std::shared_ptr<Env>*)
```
The caller should know what to do with the returned pointers.

Reply:
The entire `LDBCommand` is in `#ifndef ROCKSDB_LITE` section.

Reply:
Oh I see.

Reply:
I still find it a little weird to have two versions of `LoadEnv`. However, I don't think we can simply remove the original one. I am going to modify the newly-added one so that it will look like the following.
```
Status LoadEnv(const std::string, Env** result, std::shared_ptr<Env>* guard);
```
Then we can remove the original one during 7.0 release.
Does this sound good?

Reply:
`Status LoadEnv(const std::string, Env** result, std::shared_ptr<Env>* guard)` is definitely a better API than what we we have here.